### DISCUSSION TITLE
Is it possible to remove/purge a package using apt-offline?

### DISCUSSION THREAD

**Discussion by kyp44** [rickysarraf/apt-offline#211] [Q&A]
Hi, I am trying to purge old kernels on an offline machine using `apt purge`, but for some reason this needs an Internet connection. Is there a way to accomplish this using apt-offline?

**Comment by rickysarraf** [comment]
No. `apt-offline`  will not do it.  I'm not sure of your problem but requiring internet connectivity for removing a package sounds like a local problem.

What package is it ? Is it from stock Debian or any other proprietary vendor ? There are ways to do it but that's beyond the scope of `apt-offline` and I don't want to put the advise here as it has the potential to break your system,

**Comment by kyp44** [comment]
Many times updating my offline machine (Ubuntu 20.04) using `apt-offline` the new kernel packages are held back, so I install them using `apt-offline`. Because of this they get marked as manually installed so that `apt autoremove` will not remove them when they are no longer needed. As a result, I have about 20-30 old kernels on that machine. Evidently the way to remove them is to do `apt purge`, but for some reason `apt` is wanting to go out on the Internet to do this.
